android/tangram/src/com/mapzen/tangram/MapData.java

```diff
@@ -0,0 +1,102 @@
+package com.mapzen.tangram;
+
+import java.util.List;
+
+public class MapData {
+
+    /**
+     * Construct a new MapData object for adding drawable data to the map
+     * @param name Name of the data source in the scene file for styling this object's data
+     */
+    public MapData(String name) {
+        this.name = name;
+        this.id = addDataSource(name);
+    }
+
+    /**
+     * Get the name of this data source
+     * @return The name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Remove all data from this source
+     * @return This object, for chaining
+     */
+    public MapData clear() {
+        clearSourceData(id);
+        return this;
+    }
+
+    /**
+     * Add geometry from a GeoJSON string to this data source
+     * @param data String of GeoJSON containing a Feature or FeatureCollection
+     * @return This object, for chaining
+     */
+    public MapData addGeoJSON(String data) {
+        addSourceGeoJSON(id, data);
+        return this;
+    }
+
+    /**
+     * Add a point geometry to this data source
+     * @param point LngLat with the coordinates of the point
+     * @return This object, for chaining
+     */
+    public MapData addPoint(LngLat point) {
```

Comment:
Thinking ahead to use cases beyond simply drawing a feature on a map, we'll want to expand the data-adding functionality to include metadata about the geometry, e.g. a name, id, or multiple properties. We'll need this when we implement feature selection so that when we return the selected feature we can attach some identifying information beyond just the geometry itself. To that end, one solution would be to allow an optional string as the final argument to `addPoint`, `addLine`, and `addPolygon` which would then be interpreted as the "properties" section of a GeoJSON feature. @ecgreb What do you think about this? Is there an approach that would make things easier for the use cases that you foresee? cc @bcamper @tallytalwar